src/compiler/checker.ts

```diff
@@ -15997,7 +15998,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             // [...X[]] is equivalent to just X[]
             return readonly ? globalReadonlyArrayType : globalArrayType;
         }
-        const memberIds = namedMemberDeclarations?.length && map(namedMemberDeclarations, node => node ? getNodeId(node) : undefined);
+        const memberIds = some(namedMemberDeclarations, isDefined) && map(namedMemberDeclarations, node => node ? getNodeId(node) : undefined);

```

Comment:
I think the helper you're looking for is `mapDefined`, which lets you write:

```suggestion
        const memberIds = mapDefined(namedMemberDeclarations, node => node ? getNodeId(node) : undefined);
```

Reply:
(GitHub is failing to apply the suggestion after multiple refreshes; I'll try adding a co-author manually in my terminal)

![Screenshot of 'this diff has recently been updated. refresh and try again'](https://user-images.githubusercontent.com/3335181/227271748-1bb6a1ac-d6b4-4f97-add5-7f924bd18a27.png)

Reply:
No worries, you don't need to credit me.